### PR TITLE
Adding support for big query capacity commitments

### PR DESCRIPTION
**Pull request by chrisst**
New resource for making big query capacity commitments

Borrowed lovingly from https://github.com/GoogleCloudPlatform/magic-modules/pull/6690
fixes: https://github.com/hashicorp/terraform-provider-google/issues/6694

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_bigquery_capacity_commitment`
```

### CODE REVIEW

File: mmv1/products/bigqueryreservation/terraform.yaml [hunk 1]
```diff
@@ -23,6 +23,16 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         primary_resource_id: "reservation"
         vars:
           name: 'my-reservation'
+  CapacityCommitment: !ruby/object:Overrides::Terraform::ResourceOverride
+    id_format: "{{name}}"
+    import_format: ["{{name}}"]
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "bigquery_reservation_capacity_commitment_basic"
```
Comment (c2thorn) [inline mmv1/products/bigqueryreservation/terraform.yaml:31]: a pattern i've seen is to have a test version (`skip_docs: true`) vs a docs version (`skip_tests: true`). Would that be reasonable here?

File: mmv1/templates/terraform/examples/bigquery_reservation_capacity_commitment_basic.tf.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+resource "google_bigquery_capacity_commitment" "<%= ctx[:primary_resource_id] %>" {
+	location   = "us-west1"
+	slot_count = 100
+	plan       = "FLEX"
+}
+
+resource "time_sleep" "wait_60_seconds" {
+	depends_on = [google_bigquery_capacity_commitment.<%= ctx[:primary_resource_id] %>]
```
Comment (c2thorn) [inline mmv1/templates/terraform/examples/bigquery_reservation_capacity_commitment_basic.tf.erb:8]: this basically makes a minute delay after creating? That's the only way to delete afterwards? Is this eventually consistent?

File: mmv1/products/bigqueryreservation/terraform.yaml (hunk 1 above)
Comment (chrisst) [reply mmv1/products/bigqueryreservation/terraform.yaml:31]: SGTM - will update

File: mmv1/products/bigqueryreservation/Reservation.yaml [hunk 3]
```diff
@@ -58,8 +58,3 @@ properties:
     default_value: 0
     description: |
       Maximum number of queries that are allowed to run concurrently in this reservation. This is a soft limit due to asynchronous nature of the system and various optimizations for small queries. Default value is 0 which means that concurrency will be automatically set based on the reservation size.
-  - !ruby/object:Api::Type::Boolean
```
Comment (c2thorn) [inline mmv1/products/bigqueryreservation/Reservation.yaml:61]: even if advised from the bigquery team this is a breaking change that we'll need to separate out into a major version. Can we start by deprecating and giving the reason in the deprecation message?

Comment (chrisst) [reply mmv1/products/bigqueryreservation/Reservation.yaml:61]: IIUC It hasn't been added yet, so it's not a breaking change to remove it

Comment (chrisst) [reply mmv1/products/bigqueryreservation/Reservation.yaml:61]: oh, never mind I removed the wrong field 😳 I meant to pull this from the new resource.

**Review by c2thorn — APPROVED** [review]

**Comment by c2thorn** [comment]
rake-test is failing because of the downgrade to ruby 3.1. Should be good to merge though.

**Pull request merged**